samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -904,6 +907,17 @@ class SamzaContainer(
     })
   }
 
+  def startDiagnostics {
+    if (containerContext.config.getDiagnosticsEnabled) {
+      val rootLogger = Logger.getRootLogger
+
+      if (rootLogger.getAppender(classOf[DiagnosticsAppender].getName) == null) {
```

Comment:
Shouldn't this condition be (rootLogger.getAppender(classOf[DiagnosticsAppender].getName) *!=* null)?